### DISCUSSION TITLE
Jaeger-collector downsampling before enqueue instead of after dequeue.

### DISCUSSION THREAD

**Discussion by jianwu** [jaegertracing/jaeger#3990] [Ideas]
Seems the current collector down sampling happens after dequeuing from the collector in-memory queue before saving to storage.  Any reason not to do it before enqueue? That will protect the in-memory queue from overflowing by the misbehaved clients and it the collect more resilient for the spike of traffic. (Less chance of overflow as fewer spans are enqueued).

**Comment by yurishkuro** [comment]
agreed, makes more sense before enqueueing
